src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -2007,3 +2008,80 @@ def compute_id_mapping(
         uid = range_size * slice_val + rid + range_min
         gid = range_size * slice_val + range_min + primary_group
         return uid, gid
+
+
+class ADDNS(object):  # pylint: disable=useless-object-inheritance
+    """ ADDNS class consists of methods to search and manage AD DNS """
+
+    def __init__(self, ad_host):
+        self.ad_host = ad_host
+
+    def get_zones(self):
+        """ Returns a list of all the forward and reverse zones  on the server excluding msdcs """
+        zones = self.ad_host.run_command('dnscmd.exe /EnumZones | grep Primary | grep -v msdcs | cut -f2 -d" "')
+        return zones.stdout_text
+
+    def print_zone(self, zone):
+        """ Prints all the contents of a zone file, takes domain.com or 1.168.192.in-addr.arpa string """
+        zone = self.ad_host.run_command(f'dnscmd.exe /zoneprint {zone} | grep -v \; | sed "s/\t/ /g" | grep e |'
+                                        f'sed "s/\   / /g"')
+        return zone.stdout_text
+
+    def find_a(self, hostname):
+        """ Searches the zone for a specific A record, returning 0 for success and 1 for failure """
+        domain = self.ad_host.realm.lower()
+        host = hostname.split(".")[0]
+        results = self.ad_host.run_command(f'dnscmd.exe /zoneprint {domain} | grep -v \; | sed "s/\t/ /g" | grep e |'
+                                           f'sed "s/\   / /g" | grep {host}', raiseonerr=False)
+        if host in results.stdout_text:
```

Comment:
I think that returning bool would make more sense True if found False if not found.
Then it would be more readable in checking the result.
```
return host in  results.stdout_text
```
Note that the code style as it is has a coding standard  violation "useless else".
```
if x:
    return x
return y
```